### ISSUE TITLE
outPrefix.align.log.txt

### ISSUE THREAD

**Issue by hengxing12** [GuoliangLi-HZAU/BatMeth2#34]
hi
I run pipeline mode, and have some errors.

report2html did not output "statistic html"，
outPrefix.align.log.txt did not be produced by alignment

cp: 无法获取'./outPrefix.log.txt' 的文件状态(stat): 没有那个文件或目录
File ./outPrefix.align.log.txt Cannot be opened ....

**Comment by ZhouQiangwei** [comment]
Hi,
.align.log.txt file is generated by calmeth, could you please check does the calmeth step is correct.

**Comment by hengxing12** [comment]
hI, Guoliang

calmeth output is error and not finished. This problem is here, https://github.com/GuoliangLi-HZAU/BatMeth2/issues/33

**Comment by ZhouQiangwei** [comment]
Hi,
I saw the error is 'double free', there is only few lines about free memory in process bam file.
I have comment out this part, and updated it to github. please download and rerun calmeth.
please let us known if this error still occurs.
mo

**Comment by hengxing12** [comment]
Hi,
now is 
sh: /data/bahengxing/tools/BatMeth2-master/bin/bmtools: 没有那个文件或目录
sh: /data/bahengxing/tools/BatMeth2-master/bin/bmtools: 没有那个文件或目录
sh: /data/bahengxing/tools/BatMeth2-master/bin/bmtools: 没有那个文件或目录
File ./AAP2.align.log.txt Cannot be opened ....

**Comment by ZhouQiangwei** [comment]
sorry, i have uploaded the wrong version of BatMeth2.cpp main program, i will fix it soon.

**Comment by ZhouQiangwei** [comment]
Hi, I am just uploaded the new version, please try it.
sorry for this again

**Comment by hengxing12** [comment]
try again and no change, still error.

**Comment by ZhouQiangwei** [comment]
Is there more process information than before?



---Original---
From: ***@***.***&gt;
Date: Thu, May 12, 2022 18:31 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [GuoliangLi-HZAU/BatMeth2] outPrefix.align.log.txt (Issue #34)




 
try again and no change, still error.

**Comment by hengxing12** [comment]
[2022:05:12 19:01:00] Processed 294069857 reads.
Print output of NW_025316788.1

close and free short mem
close and free short mem
free(): double free detected in tcache 2
已放弃 (核心已转储)

**Comment by ZhouQiangwei** [comment]
yes, i think the error is here, there is two times print of "close and free short mem".
could you please send your command to me?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "GuoliangLi-HZAU/BatMeth2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月12日(星期四) 晚上7:04
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [GuoliangLi-HZAU/BatMeth2] outPrefix.align.log.txt (Issue #34)





 
[2022:05:12 19:01:00] Processed 294069857 reads.
 Print output of NW_025316788.1
 
close and free short mem
 close and free short mem
 free(): double free detected in tcache 2
 已放弃 (核心已转储)

**Comment by ZhouQiangwei** [comment]
Hi,
I have uploaded the new version. please run calmeth and check whether there is still more than one times of "close and free short mem".




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "GuoliangLi-HZAU/BatMeth2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月12日(星期四) 晚上7:04
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [GuoliangLi-HZAU/BatMeth2] outPrefix.align.log.txt (Issue #34)





 
[2022:05:12 19:01:00] Processed 294069857 reads.
 Print output of NW_025316788.1
 
close and free short mem
 close and free short mem
 free(): double free detected in tcache 2
 已放弃 (核心已转储)

**Comment by hengxing12** [comment]
[2022:05:12 23:00:33] Processed 294069857 reads.
Print output of NW_025316788.1

close and free short mem
段错误 (核心已转储)

**Comment by hengxing12** [comment]
/data/bahengxing/tools/BatMeth2/bin/calmeth -g ../index/genomic.fna  -b AAP2.sort.bam -m AAP2

**Comment by ZhouQiangwei** [comment]
Hi,
I think it may be due to the system environment. I modified some parameters of the software compilation process. Please try again.

**Comment by hengxing12** [comment]
HI

The simple test is currently successful, but i did not see ".align.log.txt" that will be used by report2html.

mC/(C+T) {260 / 6952} = 3.739931%
mCG/(CG+TG) {220 / 312} = 70.512821%
mCHG/(CHG+THG) {14 / 1648} = 0.849515%
mCHH/(CHH+THH) {26 / 4992} = 0.520833%
Done and release memory!

**Comment by ZhouQiangwei** [comment]
great, please add -as 1 parameter

**Comment by ZhouQiangwei** [comment]
great, please add -as 1 parameter for print .align.log.txt

**Comment by DharmaBums1991** [comment]
> great, please add -as 1 parameter for print .align.log.txt
Hi，the auto cmd has added this parameter,but I still encounter this error.
![image](https://user-images.githubusercontent.com/73206153/226512022-62d1ce09-c51d-4d59-b1f6-1bd7fc910ff5.png)
#39
